src/java/org/apache/cassandra/index/sai/disk/SSTableRowIdKeyRangeIterator.java

```diff
@@ -106,7 +115,10 @@ protected PrimaryKey computeNext()
             if (rowId == PostingList.END_OF_STREAM)
                 return endOfData();
 
-            return primaryKeyMap.primaryKeyFromRowId(rowId);
+            PrimaryKey pk = primaryKeyMap.primaryKeyFromRowId(rowId);
+            if (ssTableRowIdToScoreMap != null)
+                queryContext.recordScore(pk, ssTableRowIdToScoreMap.get(rowId));
```

Comment:
perhaps we can hide `LongFloatHashMap ssTableRowIdToScoreMap;` and scores from range iterators and move it into `QueryContext -> public void onPrimaryKeyRead(PrimaryKey primaryKey, SSTableId sstableId, long sstableRowId)`

Reply:
I considered this. It would be better in terms of keeping the implementation abstracted away from the iterators. However, the cost is extra map look ups for every row touched by this part of the query. It seemed better to me to avoid the map lookups, but that could be a premature optimization.

Reply:
I found a solution that is more abstract. PTAL